src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -244,11 +247,20 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
       fetchData(cursors).map { sr =>
         val items = sr.items
         val nextCursor = sr.nextCursor
-        val shouldStopEarly = items.isEmpty || (applyTimestampTermination && items.last.lastUpdatedTime >=
-          terminationTimeStamp)
+        val shouldStopEarly = items.nonEmpty && applyTimestampTermination && items.last.lastUpdatedTime >= terminationTimeStamp
         val next =
-          (nextCursor, shouldStopEarly) match {
-            case (Some(cursor), false) =>
+          (nextCursor, items.isEmpty, shouldStopEarly) match {
+            case (Some(cursor), true, _) =>
+              if (syncCursorSaveCallbackUrl.isDefined && jobId.isDefined && cursorName.isDefined) {
+                SyncCursorCallback.lastCursorCallback(
+                  syncCursorSaveCallbackUrl.get,
+                  cursorName.get,
+                  cursor,
```

Comment:
here it could be a /query cursor in backfill case, so I guess the short ways to fix is to along with
```scala
// pagingCursor -> item -> itemWithSyncCursor (either futureSyncCursor or == pagingCursor)
projectInstance: Option[String] => InstanceDefinition => ProjectedFlexibleDataModelInstance,
```
parameter add
```scala
// pagingCursorOnFinalEmptyPage -> callbackSyncCursor (sync/not backfill -> last sync cursor and it also won't be empty, backfill -> futureSyncCursor)
pickFinalCursor: Option[String] => String
```

Reply:
Fixed by introducing the projectFinalCursor.